CORTX-29122: Updated use of CORTX Secrets (#170)
Seagate/cortx-k8s: README.md

This section contains the CORTX and third-party authentication information used to deploy CORTX on Kubernetes.

A Kubernetes Secret is used to hold the various passwords and secret keys needed by the various components.
  - If the `secrets.name` field is specified, then CORTX will create and populate this secrets object, using this specified name.  CORTX will randomly generate any fields that are not mentioned or do not have values specified.

>   - If the `secrets.name_ref` field is specified, then CORTX will expect a Secrets object to already exist with the specified name, which contains the passwords for these fields.  This allows an admin to specify passwords outside of solution.yaml

Author: `external_secrets`... I like it!  Unless @osowski objects, I'll go with that.

>   - If the `secrets.name_ref` field is specified, then CORTX will expect a Secrets object to already exist with the specified name, which contains the passwords for these fields.  This allows an admin to specify passwords outside of solution.yaml

Author: I have made this change.

> 

Reviewer: Can we add an example in a sub-bullet here for how users could create a secret with those specific fields?

Something similar to
```bash
kubectl create secret generic my-cortx-secret \
  --from-literal=kafka_admin_secret=xyz123 \
  --from-literal=consul_admin_secret=abc456 \
  ...
```

Author: Good suggestion. I will do that.

Author: Done.

| Name                                      | Description                                                                             | Default Value           |
| ----------------------------------------- | --------------------------------------------------------------------------------------- | ----------------------- |
| `secrets.name`                            | Name for the Kubernetes Secret CORTX uses to store solution-specific secrets            | `cortx-secret`          |
| `secrets.content.openldap_admin_secret`   | Administrator password for the OpenLDAP required service (deprecated)                   | `<blank>`               |

> | `secrets.content.kafka_admin_secret`      | Administrator password for the Kafka required service                                   | `<blank>`               |

Author: I went with "", similar to ports.

| `secrets.content.consul_admin_secret`     | Administrator password for the Consul required service                                  | `<blank>`               |
| `secrets.content.common_admin_secret`     | Administrator password for the CORTX common services                                    | `<blank>`               |
| `secrets.content.s3_auth_admin_secret`    | Administrator password for the S3 Auth CORTX component                                  | `<blank>`               |
| `secrets.content.csm_auth_admin_secret`   | Administrator password for the CSM Auth CORTX component                                 | `<blank>`               |

> | `secrets.content.csm_mgmt_admin_secret`   | Administrator password for the CSM Management CORTX component                           | `<blank>`               |

Author: Done.

This section contains the CORTX and third-party authentication information used to deploy CORTX on Kubernetes.

A Kubernetes Secret is used to hold the various passwords and secret keys needed by the various components.
- If the `secrets.name` field is specified, then CORTX will create and populate this Secret object, using this specified name.  For any `secrets.content` fields that are not specified or do not have a value specified, CORTX will generate a random password.

> - If the `secrets.external_secret` field is specified, then CORTX will expect a Secret object to already exist with the specified name, which contains the passwords for these fields.  This allows an admin to specify passwords outside of solution.yaml.  Note: If an `screts.external_secret` is used, then the specified Secret must define _all_ CORTX-required secrets.

Reviewer 2: ```suggestion
- If the `secrets.external_secret` field is specified, then CORTX will expect a Kubernetes Secret object to already exist with the specified name, which contains the passwords for these fields.  This allows an admin to specify passwords outside of solution.yaml.  Note: If a `secrets.external_secret` is used, then the specified Secret must define _all_ CORTX-required passwords.
```
There's a mix of terminology, both "passwords" and "secrets". I picked "password" here, you could go either way.

I would also be explicit about "Kubernetes Secret", to not confuse with the password-secrets.

Author: Agreed.  I have gone with your suggestion.

> :bulb: To create a new Secret object with admin-specified values for required CORTX secrets:

Reviewer 2: ```suggestion
:bulb: To create a new Kubernetes Secret object with admin-specified values for required CORTX passwords:
```

Author: done.

```bash
kubectl create secret generic my-cortx-secret \
  --from-literal=common_admin_secret=Password1@123 \
  --from-literal=consul_admin_secret=Password2@123 \
  --from-literal=kafka_admin_secret=Password3@123 \
  --from-literal=s3_auth_admin_secret=Password4@123 \
  --from-literal=csm_auth_admin_secret=Password5@123 \
  --from-literal=csm_mgmt_admin_secret=Password6@123
```

| Name                                      | Description                                                                             | Default Value           |
| ----------------------------------------- | --------------------------------------------------------------------------------------- | ----------------------- |
| `secrets.name`                            | Name for the Kubernetes Secret CORTX uses to store solution-specific secrets            | `cortx-secret`          |

> | `secrets.content.openldap_admin_secret`   | Administrator password for the OpenLDAP required service (deprecated)                   | `""`                    |

Reviewer 2: Were you not removing this one?

> | `secrets.content.kafka_admin_secret`      | Administrator password for the Kafka required service                                   | `""`                    |

Reviewer 2: ```suggestion
| `secrets.content.kafka_admin_secret`      | Administrator password for the Kafka required service                                   | `null`                    |
```

Author: Done.
